crates/engine/tree/src/tree/payload_validator.rs

```diff
@@ -982,9 +987,10 @@ where
         parent_hash: B256,
         state: &EngineApiTreeState<N>,
         allocated_trie_input: Option<TrieInput>,
-    ) -> ProviderResult<(TrieInput, BlockNumber)> {
-        // get allocated trie input or use a default trie input
+    ) -> ProviderResult<(TrieInputSorted, TrieInput, BlockNumber)> {
```

Comment:
why do we still need `TrieInput`? cant we remove that?

Reply:
Definitely, yes, `allocated_trie_input` should be a TrieInputSorted